lib/model.js

```diff
@@ -4188,7 +4188,8 @@ function getModelsMapForPopulate(model, docs, options) {
       foreignField = foreignField.call(doc);
     }
 
-    const localFieldPath = modelSchema.paths[localField];
+    const localFieldPathType = modelSchema._getPathType(localField);
+    const localFieldPath = localFieldPathType === 'real' ? modelSchema.paths[localField] : localFieldPathType.schema;
```

Comment:
There's an awful lot of test failures here because `localFieldPathType.schema` may not contain a `getters` array.

Reply:
Added a check for the "getters" property. This was also the cause of the test failures